api/v1/coupons.js

```diff
@@ -30,6 +32,38 @@ var createCouponForNewUser = function (req, res, next) {
 };
 exports.createCouponForNewUser = createCouponForNewUser;
 
+var getDiscountOrderValueByCouponID = function (req, res, next) {
+  var couponId = req.params.couponID;
+  var username = req.query.username;
+  var orderValue = req.query.orderValue;
+
+  var isBelongToUsers = coupon.isBelongToUsers(couponId, username);
+  var isCouponValid = coupon.isCouponValid(couponId);
+  
+  Promise.join(isBelongToUsers, isCouponValid, function (belong, valid) {
+    if (!belong && valid) return true;
+    return false;
+  }).then(function (ableToUse) {
+    if (ableToUse) {
+        return coupon.getCouponCodesByCouponCode(couponId).then(function (coupons) {
+        return coupons[0];
+      }).then(function (couponObject) {
+        var ruleType = couponObject.couponRule.type;
+        var ruleValue = couponObject.couponRule.value;
+        console.log(ruleType + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+        console.log(ruleValue + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+        console.log(orderValue + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+
+        var discountedValue = coupon.getDiscountedValue(ruleType, ruleValue, orderValue);
+        res.statusCode = 201;
```

Comment:
yes, you'd need to chain your promises. For example

```
Promise.join(A, B, functon(retA, retB){
.....
return true
}).then(function(ableToUse){
  ...
 return someValue
}).then(function(someValue){
  res.send(someValue)
}).catch(...)

```

Also you have not handle the case where coupon is not valid.